### PR TITLE
llvm: Remove unneeded libcxxabi dependencies

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

I'm pretty sure this is just there for bootstrapping Darwin for some reason; normal builds of llvm given a regular stdenv don't need it for Darwin and Linux alike. Testing now to confirm.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by matthewbauer** [comment]
Another thing i forgot about is putting cmake as native build input everywhere.

**Comment by Ericson2314** [comment]
I got these builds in progress; That one I'm happy to just put through ofborg as it's basically a no-op for native. So let's do it separately.

**Comment by Ericson2314** [comment]
Testing showed that every LLVM since 3.7 worked fine (though 6 was untested). Guess this hasn't been needed for a while!

**Comment by Ericson2314** [comment]
Trivial changes (eval wise) vs last ofborg build. Merging.

File: pkgs/development/compilers/llvm/5/llvm.nix [hunk 1]
```diff
@@ -43,7 +43,8 @@ in stdenv.mkDerivation (rec {
     ++ stdenv.lib.optional enableManpages python.pkgs.sphinx;
 
   buildInputs = [ libxml2 libffi ]
-    ++ stdenv.lib.optionals stdenv.isDarwin [ libcxxabi ];
+    # TODO(@Ericson2314): Remove next mass rebuild
+    ++ stdenv.lib.optionals (stdenv.isDarwin && stdenv.hostPlatform == stdenv.buildPlatform) [ libcxxabi ];
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/development/compilers/llvm/5/llvm.nix]: @LnL7 I am working on this for LLVM 5 too, so I guess it's retroactively tested? :/

[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/development/compilers/llvm/5/llvm.nix]: For posterity. indeed staging now has the same changes with https://github.com/NixOS/nixpkgs/pull/42048

**Pull request merged**